### PR TITLE
#10: List of stored links

### PR DESCRIPTION
**Pull request by mdrlzy**
#10

### CODE REVIEW

File: app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt [hunk 1]
```diff
@@ -11,17 +20,51 @@ import kotlin.io.path.outputStream
...
+
+
+        val newLinks = linkFolder
+            .listChildren()
+            .filter {
+                it.extension == "link" && !it.isDirectory()
+            }
+            .sortedByDescending { it.getLastModifiedTime() }
+            .minus(latest.values)
+            .take(CHUNK_SIZE)
```
Comment (kirillt) [inline app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt]: Maybe "batch" or "page" would be better word. "Chunk" sounds like part of a binary file to me.

Comment (mdrlzy) [reply app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt]: Renamed to "batch"

**Comment by kirillt** [comment]
@mdrlzy Is it intentional that only 10 latest links are visible?

File: app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt [hunk 2]
```diff
@@ -23,65 +24,59 @@ import kotlin.io.path.createTempFile
...
         zipFiles(jsonFile, link.imagePath, savePath)
     }
 
-    suspend fun loadMore(): List<Link> = withContextAndLock(
-        Dispatchers.Default,
-        loadMoreMutex
-    ) {
-        val linkFolder = preferences.getLinkFolder()
-            ?: return@withContextAndLock emptyList()
+    suspend fun loadFiles(page: Int): Pair<List<Link>, Boolean> = withContext(Dispatchers.Default) {
```
Comment (kirillt) [inline app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt]: Let's replace `Pair<List<Link>, Boolean>` with dedicated structure for the sake of readability.

**Comment by mdrlzy** [comment]
Replaced with #26

**Pull request closed without merging**